Sources/Atomic.swift

```diff
@@ -8,27 +8,37 @@
 
 import Foundation
 
-final class PosixThreadMutex: NSLocking {
-	private var mutex = pthread_mutex_t()
+internal struct PosixThreadMutex {
```

Comment:
It is usually used with an apparent owner. So defining it as a struct gives the flexibility to have it embedded in the containing type, should it be a stored property, and one less pointer to chase.

> Drop precondition in favour of if-fatalError.
> 0.188 sec (2%)
>
> Embed PosixThreadMutex into Atomic.
> 0.179 sec (4%)